### PR TITLE
[DPE-8774] docs(stereo-mode): add two-node HA explanation and deployment guide

### PR DESCRIPTION
**Pull request by marceloneppel**
## Issue

Stereo mode — two-node high availability where two `postgresql` units are paired with a standalone `postgresql-watcher` that supplies a third Raft vote — shipped in `16/edge` with no user-facing documentation. There was no explanation of why the extra vote is needed and no guide for deploying the topology.

## Solution

Document stereo mode: a two-`postgresql`-unit topology paired with a standalone `postgresql-watcher` that contributes a third Raft vote, so high availability works with only two database copies.

- `explanation/stereo-mode.md`: why an odd voter count is needed, how the watcher joins the Raft quorum (`patroni-raft-controller` on port 2222), when it votes vs. abstains, and availability-zone placement — with an architecture diagram.
- `how-to/deploy/stereo-mode.md`: deploy the two units and the watcher, integrate over `watcher-offer`/`watcher`, and verify with a real `juju status` sample.
- Wire both pages into the explanation, deploy, and home-page navigation.
- Add "systemd" to the spelling wordlist.

Scoped to a single cluster. Stereo mode currently ships only in `16/edge`; a TODO marks the deploy guide for a `16/stable` update once released.

**Out of scope (intentional):** this pass covers only the single-cluster topology through the Explanation and How-to Diátaxis categories. It deliberately does not document:
- the future multi-cluster watcher architecture (one watcher serving many clusters);
- the `get-cluster-status`, `show-topology`, and `trigger-health-check` actions, which do not exist yet.

**Branch strategy:** This PR targets the `stereo-mode-docs` branch first. From there we'll decide whether these docs can be merged into `16/edge`, or whether we need to wait for the VM and K8s charm docs unification before doing so.

## Checklist
- [x] I have added or updated any relevant documentation.
- [ ] I have cleaned any remaining cloud resources from my accounts.

### CODE REVIEW

File: docs/explanation/stereo-mode.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+```{note}
+The watcher participates in **leader election only**. It stores no table data and never becomes a PostgreSQL primary or replica.
+```
+
+## When the watcher votes
+
+The watcher adjusts its vote to the number of `postgresql` units, withholding it only when the cluster already forms a safe odd majority on its own:
+
+* **An even number of units (2, 4, 6, …):** the watcher votes, making the total odd — three voters for two units, five for four — so a single failure still leaves a clear majority.
+* **An odd number of units, three or more (3, 5, 7, …):** the watcher abstains, because the `postgresql` units already form an odd voter count.
```
Comment (dragomirp) [inline docs/explanation/stereo-mode.md]: On odd numbers the watcher stops itself.

**Review by dragomirp — DISMISSED** [review]

File: docs/how-to/deploy/stereo-mode.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+(deploy-stereo-mode)=
+# How to deploy in stereo mode (two-node HA)
+
+This guide shows how to deploy Charmed PostgreSQL in **stereo mode**: two `postgresql` units for the database, plus a separate `postgresql-watcher` application that supplies a third Raft vote. This gives you high availability with only two database copies.
+
+For background on why the third vote is needed, see {ref}`stereo-mode`.
+
+```{note}
+Stereo mode is currently available in the `16/edge` channel only — it is not yet part of `16/stable`. The commands below use `16/edge` accordingly.
```
Comment (taurus-forever) [inline docs/how-to/deploy/stereo-mode.md]: I normally write: available in revision 1150+ (to avoid updating docs later after edge->stable promotion).

File: docs/how-to/deploy/stereo-mode.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+<!-- TODO: Stereo mode is currently available in the `16/edge` channel only. When the `watcher-offer` feature is released to `16/stable`, update the `--channel 16/edge` flags in the deploy commands below, the Channel column in the `juju status` sample, and the channel note above to `16/stable`. -->
```
Comment (taurus-forever) [inline docs/how-to/deploy/stereo-mode.md]: del?

File: docs/how-to/deploy/stereo-mode.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+```shell
+juju deploy postgresql-watcher pg-watcher --channel 16/edge
+```
+
+## Integrate the watcher with PostgreSQL
+
+Relate the two applications over the watcher endpoints:
+
+```shell
+juju integrate postgresql:watcher-offer pg-watcher:watcher
```
Comment (taurus-forever) [inline docs/how-to/deploy/stereo-mode.md:42]: the last chance to rename this?
Does PG really offer watcher?

Maybe:
```
juju integrate postgresql:watcher pg-watcher:watcher
```

?

P.S. we can keep it AS IS, but it looks confusing to me.

**Review by taurus-forever — COMMENTED** [review]
Nice!!!

File: docs/how-to/deploy/stereo-mode.md (hunk 4 above)
Comment (marceloneppel) [reply docs/how-to/deploy/stereo-mode.md:42]: I agree. It's also confusing to me now that you commented about it. We can fix in the charm, but it will need a new solution QA batch of tests on a new candidate after we merge to 16/edge, right? I mean, do we have time for that until the release?

File: docs/how-to/deploy/stereo-mode.md (hunk 2 above)
Comment (marceloneppel) [reply docs/how-to/deploy/stereo-mode.md]: Fixed on https://github.com/canonical/postgresql-operator/pull/1740/changes/52229eca6966237361a14ee8acfdbc72552f9f95.

File: docs/how-to/deploy/stereo-mode.md (hunk 3 above)
Comment (marceloneppel) [reply docs/how-to/deploy/stereo-mode.md]: Removed on https://github.com/canonical/postgresql-operator/pull/1740/changes/52229eca6966237361a14ee8acfdbc72552f9f95.

File: docs/explanation/stereo-mode.md (hunk 1 above)
Comment (marceloneppel) [reply docs/explanation/stereo-mode.md]: Updated on https://github.com/canonical/postgresql-operator/pull/1740/changes/52229eca6966237361a14ee8acfdbc72552f9f95.

**Review by a-velasco — DISMISSED** [review]
This looks great @marceloneppel, thank you :) Let's merge it here so that it's available in the current docs for the release. 

I'll then integrate it manually into my branch for the single kernel migration.

File: docs/how-to/deploy/stereo-mode.md (hunk 4 above)
Comment (taurus-forever) [reply docs/how-to/deploy/stereo-mode.md:42]: Discussed with Mohamed on sync call.
1) we can go AS IS. 
2) IF Mohamed want to change endpoint name, it is today only. There will be a message
3) we can merge docs, as IF changes are necessary, both docs and charm code will require changes.

**Review by taurus-forever — DISMISSED** [review]

**Review by a-velasco — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]

**Review by dragomirp — APPROVED** [review]
